src/evaluator/fallbackTreatmentsCalculator/index.ts

```diff
@@ -0,0 +1,61 @@
+import { FallbackTreatmentConfiguration, FallbackTreatment } from '../../../types/splitio';
+import { FallbacksSanitizer } from './fallbackSanitizer';
+import { CONTROL } from '../../utils/constants';
+import { isString } from '../../utils/lang';
+import { ILogger } from '../../logger/types';
+
+export type IFallbackTreatmentsCalculator = {
+  resolve(flagName: string, label?: string): FallbackTreatment & { label?: string };
+}
+
+export class FallbackTreatmentsCalculator implements IFallbackTreatmentsCalculator {
+  private readonly labelPrefix = 'fallback - ';
+  private readonly fallbacks: FallbackTreatmentConfiguration;
+
+  constructor(logger: ILogger, fallbacks?: FallbackTreatmentConfiguration) {
+    const sanitizedGlobal = fallbacks?.global ? FallbacksSanitizer.sanitizeGlobal(logger, fallbacks.global) : undefined;
+    const sanitizedByFlag = fallbacks?.byFlag ? FallbacksSanitizer.sanitizeByFlag(logger, fallbacks.byFlag) : {};
+    this.fallbacks = {
+      global: sanitizedGlobal,
+      byFlag: sanitizedByFlag
+    };
+  }
+
+  resolve(flagName: string, label?: string): FallbackTreatment & { label?: string } {
```

Comment:
We are calling `resolve` always with a label. So, we can make it mandatory to simplify some code:
```suggestion
  resolve(flagName: string, label: string): TreatmentWithConfig & { label: string } {
```